src/main/java/net/snowflake/client/jdbc/SnowflakePreparedStatementV1.java

```diff
@@ -298,6 +298,12 @@ public void setTime(int parameterIndex, Time x) throws SQLException {
       // Convert to nanoseconds since midnight using the input time mod 24 hours.
       final long MS_IN_DAY = 86400 * 1000;
       long msSinceEpoch = x.getTime();
+      if (msSinceEpoch < 0) {
+        throw new SnowflakeSQLLoggedException(
+            connection.getSfSession(),
+            SqlState.INVALID_PARAMETER_VALUE,
+            "Invalid bind value " + msSinceEpoch + " for type TIME");
+      }
       // Use % + % instead of just % to get the nonnegative remainder.
       // TODO(mkember): Change to use Math.floorMod when Client is on Java 8.
       long msSinceMidnight = (msSinceEpoch % MS_IN_DAY + MS_IN_DAY) % MS_IN_DAY;
```

Comment:
Previously, any negative millisecond value from a negative Time object would be converted into a positive value with this line: `long msSinceMidnight = (msSinceEpoch % MS_IN_DAY + MS_IN_DAY) % MS_IN_DAY;` So the SQL compilation exception was never hit since time values were always positive. Now, I've added an exception above this line so that negative values throw their own exception.